CN panic: unexpected external call to newEpochNextValidators cache

## Current Behavior
```
panic: bug: unexpected external call to newEpochNextValidators cache
goroutine 297 [running]:
github.com/nspcc-dev/neo-go/pkg/core/native.(*NEO).ComputeNextBlockValidators(0x20?, 0x2?)
        github.com/nspcc-dev/neo-go/pkg/core/native/native_neo.go:1132 +0x68
github.com/nspcc-dev/neo-go/pkg/core.(*Blockchain).ComputeNextBlockValidators(0xc001824478?)
        github.com/nspcc-dev/neo-go/pkg/core/blockchain.go:2736 +0x25
github.com/nspcc-dev/neo-go/pkg/consensus.(*service).getValidators(0xc000859d40, {0xc0004b2800?, 0x11392c0?, 0xc0036a9620?})
        github.com/nspcc-dev/neo-go/pkg/consensus/consensus.go:688 +0x42
github.com/nspcc-dev/dbft.(*Context).Fill(0xc00028f8c0)
        github.com/nspcc-dev/dbft@v0.0.0-20230515113611-25db6ba61d5c/context.go:268 +0x222
github.com/nspcc-dev/dbft.(*Context).makePrepareRequest(0xc00028f8c0)
        github.com/nspcc-dev/dbft@v0.0.0-20230515113611-25db6ba61d5c/send.go:19 +0x1c
github.com/nspcc-dev/dbft.(*DBFT).sendPrepareRequest(0xc00028f8c0)
        github.com/nspcc-dev/dbft@v0.0.0-20230515113611-25db6ba61d5c/send.go:31 +0x29
github.com/nspcc-dev/dbft.(*DBFT).OnTimeout(0xc00028f8c0, {0x11e2a2b?, 0x0?})
        github.com/nspcc-dev/dbft@v0.0.0-20230515113611-25db6ba61d5c/dbft.go:190 +0x3d5
github.com/nspcc-dev/neo-go/pkg/consensus.(*service).eventLoop(0xc000859d40)
        github.com/nspcc-dev/neo-go/pkg/consensus/consensus.go:314 +0x368
created by github.com/nspcc-dev/neo-go/pkg/consensus.(*service).Start in goroutine 116
        github.com/nspcc-dev/neo-go/pkg/consensus/consensus.go:285 +0x125
```

## Expected Behavior
No panic.

## Possible Solution
Fix the damn thing!

## Steps to Reproduce
Run CN. Nothing else known, but likely some votes and committee changes are needed.

## Your Environment
* Version of the product used: 0.104.0, linux-amd64

--- a/pkg/consensus/consensus.go
+++ b/pkg/consensus/consensus.go
@@ -681,12 +681,12 @@ func (s *service) getValidators(txes ...block.Transaction) []crypto.PublicKey {
 		// block's validators, thus should return validators from the current
 		// epoch without recalculation.
 		pKeys, err = s.Chain.GetNextBlockValidators()
-	} else {
-		// getValidators with non-empty args is used by dbft to fill block's
-		// NextConsensus field, ComputeNextBlockValidators will return proper
-		// value for NextConsensus wrt dBFT epoch start/end.
-		pKeys = s.Chain.ComputeNextBlockValidators()
 	}
+	// getValidators with non-empty args is used by dbft to fill block's
+	// NextConsensus field, but NeoGo doesn't provide WithGetConsensusAddress
+	// callback and fills NextConsensus by itself via WithNewBlockFromContext
+	// callback. Thus, leave pKeys empty if txes != nil.
+
 	if err != nil {
 		s.log.Error("error while trying to get validators", zap.Error(err))
 	}
@@ -727,6 +727,16 @@ func (s *service) newBlockFromContext(ctx *dbft.Context) block.Block {
 		block.PrevStateRoot = sr.Root
 	}
 
+	// ComputeNextBlockValidators returns proper set of validators wrt dBFT epochs
+	// boundary. I.e. for the last block in the dBFT epoch this method returns the
+	// list of validators recalculated from the latest relevant information about
+	// NEO votes; in this case list of validators may differ from the one returned
+	// by GetNextBlockValidators. For the not-last block of dBFT epoch this method
+	// returns the same list as GetNextBlockValidators. Note, that by this moment
+	// we must be sure that previous block was successfully persisted to chain
+	// (i.e. PostPersist was completed for native Neo contract and PostPersist
+	// execution cache was persisted to s.Chain's DAO), otherwise the wrong
+	// (outdated, relevant for the previous dBFT epoch) value will be returned.
 	var validators = s.Chain.ComputeNextBlockValidators()
 	script, err := smartcontract.CreateDefaultMultiSigRedeemScript(validators)
 	if err != nil {

--- a/pkg/core/native/native_neo.go
+++ b/pkg/core/native/native_neo.go
@@ -971,7 +971,6 @@ func (n *NEO) ModifyAccountVotes(acc *state.NEOBalance, d *dao.Simple, value *bi
 				return nil
 			}
 		}
-		cache.newEpochNextValidators = nil
 		return putConvertibleToDAO(n.ID, d, key, cd)
 	}
 	return nil

